internal/input/report_test.go

```diff
@@ -0,0 +1,469 @@
+		// 				TotalViolations: 1,
+		// 				TotalWarnings:   1,
+		// 				TotalSuccesses:  1,
+		// 				Success:         false,
+		// 				Name:            "",
+		// 			},
+		// 		},
+		// 		Success: false,
+		// 		Key:     utils.TestPublicKey,
+		// 	},
+		// },
+		// {
+		// 	name:     "with snapshot",
+		// 	snapshot: "snappy",
+		// 	input: Component{
+		// 		Violations: []evaluator.Result{
+		// 			{
+		// 				Message: "violation",
+		// 				Metadata: map[string]interface{}{
+		// 					"code": "violation",
+		// 				},
+		// 			},
+		// 		},
+		// 		Warnings: []evaluator.Result{
+		// 			{
+		// 				Message: "warning",
+		// 				Metadata: map[string]interface{}{
+		// 					"code": "warning",
+		// 				},
+		// 			},
+		// 		},
+		// 		Successes: []evaluator.Result{
+		// 			{
+		// 				Message: "success",
+		// 				Metadata: map[string]interface{}{
+		// 					"code": "success",
+		// 				},
+		// 			},
+		// 		},
+		// 		Success:      false,
+		// 		SuccessCount: 1,
+		// 	},
+		// 	want: summary{
+		// 		Snapshot: "snappy",
+		// 		Components: []componentSummary{
+		// 			{
+		// 				Violations:      map[string][]string{"violation": {"violation"}},
+		// 				Warnings:        map[string][]string{"warning": {"warning"}},
+		// 				Successes:       map[string][]string{"success": {"success"}},
+		// 				TotalViolations: 1,
+		// 				TotalWarnings:   1,
+		// 				TotalSuccesses:  1,
+		// 				Success:         false,
+		// 				Name:            "",
+		// 			},
+		// 		},
+		// 		Success: false,
+		// 		Key:     utils.TestPublicKey,
+		// 	},
+		// },
+	}
+
+	for _, tc := range tests {
+		t.Run(fmt.Sprintf("NewReport=%s", tc.name), func(t *testing.T) {
+			ctx := context.Background()
+			report, err := NewReport(tc.input, createTestPolicy(t, ctx), "data here", nil)
+			// report, err := NewReport(tc.snapshot, []Component{tc.input}, createTestPolicy(t, ctx), "data here", nil)
+			assert.NoError(t, err)
+			fmt.Println("\n\nExpected:\n", tc.want, "\n\nActual:\n", report.toSummary())
+			assert.Equal(t, tc.want, report.toSummary())
+		})
+	}
+
+}
+
+func testComponentsFor(filePaths []string) []Input {
+	inputs := []Input{
+		{
+			FilePath: filePaths[0],
+			Violations: []evaluator.Result{
+				{
+					Message: "violation1",
+				},
+			},
+			Warnings: []evaluator.Result{
+				{
+					Message: "warning1",
+				},
+			},
+			Successes: []evaluator.Result{
+				{
+					Message: "success1",
+				},
+			},
+			Success: false,
+		},
+		{
+			FilePath: filePaths[1],
+			Violations: []evaluator.Result{
+				{
+					Message: "violation2",
+				},
+			},
+			Success: false,
+		},
+		{
+			FilePath: filePaths[2],
+			Successes: []evaluator.Result{
+				{
+					Message: "success3",
+				},
+			},
+			Success: true,
+		},
+	}
+	return inputs
+}
+
+func createTestPolicy(t *testing.T, ctx context.Context) policy.Policy {
+	utils.SetTestRekorPublicKey(t)
+
+	p, err := policy.NewPolicy(ctx, policy.Options{
+		PublicKey:     utils.TestPublicKey,
```

Comment:
This shouldn't be required here.

Reply:
Ack. Updated to use proper policy type which removes this field.